sycl/test-e2e/Basic/buffer/subbuffer.cpp

```diff
@@ -234,8 +234,10 @@ void copyBlock() {
 
   auto CopyF = [](buffer &Buffer, buffer &Block, size_t Idx, size_t BlockSize) {
     auto Subbuf = buffer(Buffer, Idx * BlockSize, BlockSize);
-    auto *Src = Subbuf.get_access<mode::read>().get_pointer();
-    auto *Dst = Block.get_access<mode::write>().get_pointer();
+    sycl::host_accessor SrcAcc(Subbuf, sycl::read_only);
+    sycl::host_accessor DstAcc(Block, sycl::write_only);
+    auto *Src = SrcAcc.get_pointer();
+    auto *Dst = DstAcc.get_pointer();
```

Comment:
I think code on the left might be an UB where we just got lucky. The changes makes sure the accessors are alive when the pointers are used.